app/main.py

```diff
@@ -1,3 +1,61 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        new_distance = Distance(self.km)
+        if isinstance(other, (int, float)):
+            new_distance.km += other
+        elif isinstance(other, Distance):
+            new_distance.km += other.km
+        return new_distance
+
+    def __iadd__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, (int, float)):
+            self.km += other
+        elif isinstance(other, Distance):
+            self.km += other.km
+        return self
+
+    def __mul__(self, multiplier: int) -> Distance:
+        return Distance(self.km * multiplier)
+
+    def __truediv__(self, divider: int) -> Distance:
+        divided_distance = Distance(self.km)
+        if divider != 0:
+            divided_distance.km = round(divided_distance.km / divider, 2)
+        return divided_distance
```

Comment:
No, it can be int | float. But it can't be an instance of Distance, so please add a check that  multiplier and divider are not instances of Distance